clap_derive/src/utils/doc_comments.rs

```diff
@@ -117,10 +83,200 @@ fn is_blank(s: &str) -> bool {
     s.trim().is_empty()
 }
 
-fn merge_lines(lines: impl IntoIterator<Item = impl AsRef<str>>) -> String {
-    lines
-        .into_iter()
-        .map(|s| s.as_ref().trim().to_owned())
-        .collect::<Vec<_>>()
-        .join(" ")
+mod markdown {
+    use anstyle::{Reset, Style};
+    use pulldown_cmark::{Event, Options, Parser, Tag, TagEnd};
+    use std::fmt;
+    use std::fmt::Write;
+    use std::ops::AddAssign;
+
+    #[derive(Default)]
+    struct MarkdownWriter {
+        output: String,
+        indentation: usize,
+        styles: Vec<Style>,
+    }
+
+    impl MarkdownWriter {
+        fn newline(&mut self) {
+            self.output.push('\n');
+        }
+        fn endline(&mut self) {
+            if !self.output.ends_with('\n') {
+                self.newline();
+            }
+        }
+
+        fn write_fmt(&mut self, arguments: fmt::Arguments<'_>) {
+            if self.output.ends_with('\n') {
+                write!(self.output, "{0: <1$}", "", self.indentation).unwrap();
+            }
+            self.output.write_fmt(arguments).unwrap();
+        }
+
+        fn start_link(&mut self, dest_url: pulldown_cmark::CowStr<'_>) {
+            write!(self, "\x1B]8;;{dest_url}\x1B\\");
+        }
+        fn end_link(&mut self) {
+            write!(self, "\x1B]8;;\x1B\\");
+        }
+
+        fn start_style(&mut self, style: Style) {
+            self.styles.push(style);
+            write!(self, "{style}");
+        }
+        fn end_style(&mut self, style: Style) {
+            let last_style = self.styles.pop();
+            debug_assert_eq!(last_style.unwrap(), style);
+
+            write!(self, "{Reset}");
+            // Reapplying all, because anstyle doesn't support merging styles
+            // (probably because the ambiguity around colors)
+            // TODO If we decide not to support any colors, we can replace this with
+            // anstyle::Effects and remove the need for applying them all individually.
+            for style in &self.styles {
+                write!(self.output, "{style}").unwrap();
+            }
+        }
+    }
+
+    pub(super) fn parse_markdown(input: &[String]) -> (String, Option<String>) {
+        // Markdown Configuration
+        let parsing_options = Options::ENABLE_STRIKETHROUGH /* TODO UNICODE | Options::ENABLE_SMART_PUNCTUATION */;
+        // Minimal Styling for now, because we cannot configure it
+        let style_heading = Style::new().bold();
+        let style_emphasis = Style::new().italic();
+        let style_strong = Style::new().bold();
+        let style_strike_through = Style::new().strikethrough();
+        let style_link = Style::new().underline();
+        // TODO decide how to style code
+        let style_code = Style::new().dimmed();
```

Comment:
Ideally this would be in the `literal` style but we don't have access to it

Reply:
Should I leave dimmed for now? Or remove it and just emit `` ` ``?

Reply:
Dimmed is for de-emphasizing which is the opposite of what we want for literals.  I've also heard reports that dimmed is handled poorly by some terminals.

Maybe invert?

Reply:
> Maybe invert?

IIRC, invert also had some problems sometimes, but we could go for it for now, as you are right, de-emphasizing is not what we should be doing.

Really makes me think that this is only fully completed once we find a way to use the correct styles.

Would there be any value in using the default styles?